drivers/flash/flash_stm32_qspi.c

```diff
@@ -695,14 +694,9 @@ static int flash_stm32_qspi_init(const struct device *dev)
 	struct dma_config dma_cfg = dev_data->dma.cfg;
 	static DMA_HandleTypeDef hdma;
 
-	if (dev_data->dma.name != NULL) {
-		dev_data->dma.dev = device_get_binding(dev_data->dma.name);
-		if (!dev_data->dma.dev) {
-			LOG_ERR("%s device not found", dev_data->dma.name);
-			return -ENODEV;
-		}
-	} else {
-		return -EINVAL;
+	if (!device_is_ready(dev_data->dma.dev)) {
+		LOG_ERR("%s device not found", dev_data->dma.dev->name);
```

Comment:
Message should probably be `device is not ready` or something like that.

Reply:
Right, fixed